contracts/0.8.9/oracle/AccountingOracle.sol

```diff
@@ -0,0 +1,817 @@
+    error ExitedValidatorsLimitExceeded(uint256 limitPerDay, uint256 exitedPerDay);
+    error UnsupportedExtraDataFormat(uint256 format);
+    error UnsupportedExtraDataType(uint256 dataType);
+    error MaxExtraDataItemsCountExceeded(uint256 maxItemsCount, uint256 receivedItemsCount);
+    error CannotSubmitExtraDataBeforeMainData();
+    error ExtraDataAlreadyProcessed();
+    error ExtraDataListOnlySupportsSingleTx();
+    error UnexpectedExtraDataFormat(uint256 expectedFormat, uint256 receivedFormat);
+    error UnexpectedExtraDataItemsCount(uint256 expectedCount, uint256 receivedCount);
+    error UnexpectedExtraDataIndex(uint256 expectedIndex, uint256 receivedIndex);
+    error InvalidExtraDataSortOrder();
+
+    event DataBoundariesSet(uint256 maxExitedValidatorsPerDay, uint256 maxExtraDataListItemsCount);
+    event ExtraDataSubmitted(uint256 indexed refSlot, uint256 itemsProcessed, uint256 itemsCount);
+
+    event WarnExtraDataIncompleteProcessing(
+        uint256 indexed refSlot,
+        uint256 processedItemsCount,
+        uint256 itemsCount
+    );
+
+    struct DataBoundaries {
+        uint64 maxExitedValidatorsPerDay;
+        uint64 maxExtraDataListItemsCount;
+    }
+
+    struct ExtraDataProcessingState {
+        uint64 refSlot;
+        uint16 dataFormat;
+        uint64 itemsCount;
+        uint64 itemsProcessed;
+        uint256 lastProcessedItem;
+        bytes32 dataHash;
+    }
+
+    /// @notice An ACL role granting the permission to submit the data for a committee report.
+    bytes32 public constant SUBMIT_DATA_ROLE = keccak256("SUBMIT_DATA_ROLE");
+
+    /// @notice An ACL role granting the permission to set report data safety boundaries.
+    bytes32 constant public MANAGE_DATA_BOUNDARIES_ROLE = keccak256("MANAGE_DATA_BOUNDARIES_ROLE");
+
+
+    /// @dev Storage slot: DataBoundaries dataBoundaries
+    bytes32 internal constant DATA_BOUNDARIES_POSITION =
+        keccak256("lido.AccountingOracle.dataBoundaries");
+
+    /// @dev Storage slot: ExtraDataProcessingState state
+    bytes32 internal constant EXTRA_DATA_PROCESSING_STATE_POSITION =
+        keccak256("lido.AccountingOracle.extraDataProcessingState");
+
+    /// @dev Storage slot: address legacyOracle
+    bytes32 internal constant LEGACY_ORACLE_POSITION =
+        keccak256("lido.AccountingOracle.legacyOracle");
+
+
+    address public immutable LIDO;
+    IComponentLocator public immutable LOCATOR;
+
+    ///
+    /// Initialization & admin functions
+    ///
+
+    constructor(address componentLocator, uint256 secondsPerSlot, uint256 genesisTime)
+        BaseOracle(secondsPerSlot, genesisTime)
+    {
+        if (componentLocator == address(0)) revert ComponentLocatorCannotBeZero();
+        LOCATOR = IComponentLocator(componentLocator);
+        LIDO = LOCATOR.lido();
+    }
+
+    function initialize(
+        address admin,
+        address consensusContract,
+        uint256 consensusVersion,
+        address legacyOracle,
```

Comment:
`legacyOracle` can be fetched from `LOCATOR`

Reply:
initialization was refactored